### ISSUE TITLE
DOC102 false negative for NumPy-style comma-separated parameters

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#20959]
### Summary

[The NumPy docstring style guide says](https://numpydoc.readthedocs.io/en/stable/format.html#parameters) “When two or more input parameters have exactly the same type, shape and description, they can be combined”, but [`docstring-extraneous-parameter` (DOC102)](https://docs.astral.sh/ruff/rules/docstring-extraneous-parameter/) doesn’t support that, which causes false negatives. [Example](https://play.ruff.rs/118cf9ce-c72c-4cd8-b967-42d65e646f68):
```console
$ cat >doc102.py <<'# EOF'
def leq(x: object, y: object) -> bool:
    """Compare two objects for loose equality.

    Parameters
    ----------
    x1, x2 : object
        Objects.

    Returns
    -------
    bool
        Whether the objects are identical or equal.
    """
    return x is y or x == y
# EOF

$ ruff --isolated check doc102.py --select DOC102 --preview
All checks passed!
```

### Version

ruff 0.14.1 (2bffef596 2025-10-16)
